src/Features/CSharp/Portable/SignatureHelp/PrimaryConstructorBaseTypeSignatureHelpProvider.cs

```diff
@@ -0,0 +1,153 @@
+
+            var currentConstructor = semanticModel.GetSymbolInfo(baseTypeSyntax, cancellationToken).Symbol;
+            var selectedItem = TryGetSelectedIndex(accessibleConstructors, currentConstructor);
+
+            return CreateSignatureHelpItems(accessibleConstructors.SelectAsArray(c =>
+                Convert(c, baseTypeSyntax.ArgumentList.OpenParenToken, semanticModel, anonymousTypeDisplayService, documentationCommentFormattingService)).ToList(),
+                textSpan, GetCurrentArgumentState(root, position, syntaxFacts, textSpan, cancellationToken), selectedItem);
+        }
+
+        public override SignatureHelpState GetCurrentArgumentState(SyntaxNode root, int position, ISyntaxFactsService syntaxFacts, TextSpan currentSpan, CancellationToken cancellationToken)
+        {
+            if (TryGetBaseTypeSyntax(root, position, syntaxFacts, SignatureHelpTriggerReason.InvokeSignatureHelpCommand, cancellationToken, out var expression) &&
+                currentSpan.Start == SignatureHelpUtilities.GetSignatureHelpSpan(expression.ArgumentList).Start)
+            {
+                return SignatureHelpUtilities.GetSignatureHelpState(expression.ArgumentList, position);
+            }
+
+            return null;
+        }
+
+        private static SignatureHelpItem Convert(
+            IMethodSymbol constructor,
+            SyntaxToken openToken,
+            SemanticModel semanticModel,
+            IAnonymousTypeDisplayService anonymousTypeDisplayService,
+            IDocumentationCommentFormattingService documentationCommentFormattingService)
+        {
+            var position = openToken.SpanStart;
+            var item = CreateItem(
+                constructor, semanticModel, position,
+                anonymousTypeDisplayService,
+                constructor.IsParams(),
+                constructor.GetDocumentationPartsFactory(semanticModel, position, documentationCommentFormattingService),
+                GetPreambleParts(constructor, semanticModel, position),
+                GetSeparatorParts(),
+                GetPostambleParts(),
+                constructor.Parameters.Select(p => Convert(p, semanticModel, position, documentationCommentFormattingService)).ToList());
+            return item;
+
+            static IList<SymbolDisplayPart> GetPreambleParts(
+                IMethodSymbol method,
+                SemanticModel semanticModel,
+                int position)
+            {
+                var result = new List<SymbolDisplayPart>();
+
+                result.AddRange(method.ContainingType.ToMinimalDisplayParts(semanticModel, position));
+                result.Add(Punctuation(SyntaxKind.OpenParenToken));
+
+                return result;
+            }
+
+            static IList<SymbolDisplayPart> GetPostambleParts()
+            {
+                return SpecializedCollections.SingletonList(Punctuation(SyntaxKind.CloseParenToken));
+            }
```

Comment:
i'd just inline this myself.

Reply:
I prefer to keep this as-is to match other SignatureHelp providers. There's opportunity to factor such methods out.